Tasks/SonarQubePostTest/SonarQubePostTest.ps1

```diff
@@ -34,4 +34,23 @@ else
      Write-Warning "Could not find the summary report file $summaryMdPath"
 }
 
+. .\CodeAnalysisFilePathComputation.ps1
+$prcaEnabled = Get-TaskVariable -Context $distributedTaskContext -Name "GitPullRequestCodeAnalysisEnabled"
+if ($prcaEnabled -ieq "true")
+{
+    ComputeCodeAnalysisFilePaths $agentBuildDirectory
+
+    $sonarReportFolderPath = [System.IO.Path]::Combine($agentBuildDirectory, ".sonarqube", "out", ".sonar")
+    $sonarReportFilePathProcessed = [System.IO.Path]::Combine($sonarReportFolderPath, "code-analysis-report.json")
 
+
+    if ([System.IO.File]::Exists($sonarReportFilePathProcessed))
+    {
+        Write-Verbose -Verbose "Uploading build artifact code-analysis-report.json from $sonarReportFolderPath"
+        Write-Host "##vso[artifact.upload containerfolder=CodeAnalysisIssues;artifactname=CodeAnalysisIssues;]$sonarReportFilePathProcessed"
+    }
+    else
+    {
+        Write-Warning "Could not find code-analysis-report.json file in $sonarReportFilePathProcessed"
+    }
```

Comment:
Why not push all of the logic including the upload to into the CodeAnalysisFilePathComputation.ps1?
That would centralise as much of the PR logic as possible in one place and would avoid the need to recompute the file path to the created file.